README.md

```diff
@@ -1,636 +1,524 @@
+  file and the path specified in its `include` block. We use this so that each Terraform module stores its Terraform
+  state at a different `key`. For example, the `mysql` module will have its `key` parameter resolve to 
+  `mysql/terraform.tstate` and the `frontend-app` module will have its `key` parameter resolve to 
+  `frontend-app/terraform.tfstate`.
 
-Terraform itself also supports loading variables via the environment. It is possible to use the same variables by 
-correctly using the terraform prefix `TF_VAR_`.
+See [the helper functions docs](#helper-functions) for more info.
 
-```bash
-TF_VAR_variable="value" terragrunt apply
-```
+
+#### Create remote state and locking resources automatically
+
+When you run `terragrunt` with `remote_state` configuration, it will automatically create the following resources if 
+they don't already exist:
+
+* **S3 bucket**: If you are using the [S3 backend](https://www.terraform.io/docs/backends/types/s3.html) for remote 
+  state storage and the `bucket` you specify in `remote_state.config` doesn't already exist, Terragrunt will create it 
+  automatically, with [versioning enabled](http://docs.aws.amazon.com/AmazonS3/latest/dev/Versioning.html). 
+
+* **DynamoDB table**: If you are using the [S3 backend](https://www.terraform.io/docs/backends/types/s3.html) for 
+  remote state storage and you specify a `lock_table` (a [DynamoDB table used for 
+  locking](https://www.terraform.io/docs/backends/types/s3.html#lock_table)) in `remote_state.config`, if that table
+  doesn't already exist, Terragrunt will create it automatically, including a primary key called `LockID`.
+
+**Note*: If you specify a `profile` key in `remote_state.config`, Terragrunt will automatically use this AWS profile 
+when creating the S3 bucket or DynamoDB table. 
+
+
+
+
+
+
+
+### Keep your CLI flags DRY
+
+#### Motivation
+
+Sometimes you may need to pass extra CLI arguments every time you run certain `terraform` commands. For example, you 
+may want to set the `lock-timeout` setting to 20 minutes for all commands that may modify remote state so that 
+Terraform will keep trying to acquire a lock for up to 20 minutes if someone else already has the lock rather than 
+immediately exiting with an error.
+
+You can configure Terragrunt to pass specific CLI arguments for specific commands using an `extra_arguments` block
+in your `terraform.tfvars` file:
 
 ```hcl
 terragrunt = {
-  remote_state {
-    backend = "s3"
-    config {
-      encrypt = "true"
-      bucket = "${get_env("TF_VAR_variable", "value")}-bucket"
-      key = "/foo/bar/terraform.tfstate"
-      region = "us-west-2"
-    }
+  terraform {
+    # Force Terraform to keep trying to acquire a lock for up to 20 minutes if someone else already has the lock 
+    extra_arguments "retry_lock" {
+      commands = [
+        "init",
+        "apply",
+        "refresh",
+        "import",
+        "plan",
+        "taint",
+        "untaint"
+      ]
```

Comment:
> This list of terraform commands that apply to locks is non-obvious. It'd be nice if we could another helper command here that generates this array for you.

This is another good candidate for a separate issue, if you don't mind filing it.

> BTW, where does retry_lock come from? I didn't see it in Terraform docs after a quick search, so I'm guessing it's an arbitrary var name? Would be good to clarify either way.

Arbitrary name. I updated the docs to clarify: https://github.com/gruntwork-io/terragrunt/pull/167/commits/5af50b0d64c2cf113a35458edbac78085597401c